src/Object/ValidationListener.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Zenstruck\Foundry\Object;
+
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use Zenstruck\Foundry\Object\Event\AfterInstantiate;
+
+final class ValidationListener
+{
+    public function __construct(
+        private readonly ValidatorInterface $validator
+    ) {
+    }
+
+    public function __invoke(AfterInstantiate $event): void
+    {
+        if (!$event->factory->validationEnabled()) {
+            return;
+        }
+
+        $violations = $this->validator->validate($event->object);
+
+        if ($violations->count() > 0) {
+            throw new ValidationFailedException($event->object, $violations);
```

Comment:
I'm wondering if we should throw this exception or another one from Foundry's namespace.  This ones is kinda ugly:
```
Symfony\Component\Validator\Exception\ValidationFailedException: Object(Zenstruck\Foundry\Tests\Fixture\Entity\EntityForValidation).name:
    This value should not be blank. (code c1051bb4-d103-4f74-8988-acbcafc7fdc3)
```